Resync requestIdleCallback WPT tests

rniwa opened the pull request:
#### d9b63e8534ad280a80993b129eb33269dd3a8ec9
<pre>
Resync requestIdleCallback WPT tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=259917">https://bugs.webkit.org/show_bug.cgi?id=259917</a>

Reviewed by Tim Nguyen.

Updated WPT tests for requestIdleCallback as of 0f0227f189402588741ae46de5e4d62888347961.

* LayoutTests/TestExpectations: Added flaky failure expectation on the tests that yield different result each time.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/basic.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-idle-periods.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-invoked.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-timeRemaining-cross-realm-method.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-after-expired-timer-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-after-expired-timer.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-dynamic-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-dynamic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-timeout-dynamic-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-timeout-dynamic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/resources/ric-utils.js: Added.
(getDeadlineForNextIdleCallback):
(getRICRetryCount):
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/266678@main">https://commits.webkit.org/266678@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97c623d8a5b30085753e82b8afd798b43868fc45

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14443 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15095 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16182 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17268 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14829 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14622 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15159 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12260 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16906 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12444 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13023 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20025 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13521 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/13187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/16402 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13741 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11577 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13029 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/12880 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17366 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1722 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13585 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-timeRemaining-cross-realm-method.html:
```diff
@@ -1,4 +1,4 @@
-<!doctype html><!-- webkit-test-runner [ RequestIdleCallbackEnabled=true ] -->
+<!doctype html><!-- webkit-test-runner [ RequestIdleCallbackEnabled=true ] --><!-- webkit-test-runner [ RequestIdleCallbackEnabled=true ] -->
```
Is this a typo? This doesn't seem to be upstream (although it's strange that upstream has `webkit-test-runner` in the first place, but separate issue, we should probably just mark RequestIdleCallbackEnabled as testable in another PR)

rniwa replied:
Weird. Upstream has `<!-- webkit-test-runner [ RequestIdleCallbackEnabled=true ] -->` but not sure why it got duplicated here.

rniwa committed:
Resync requestIdleCallback WPT tests
https://bugs.webkit.org/show_bug.cgi?id=259917

Reviewed by Tim Nguyen.

Updated WPT tests for requestIdleCallback as of 0f0227f189402588741ae46de5e4d62888347961.

* LayoutTests/TestExpectations: Added flaky failure expectation on the tests that yield different result each time.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/basic.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-idle-periods.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-invoked.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/callback-timeRemaining-cross-realm-method.html:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-after-expired-timer-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-after-expired-timer.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-dynamic-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-dynamic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-rAF.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-timeout-dynamic-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max-timeout-dynamic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/deadline-max.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/resources/ric-utils.js: Added.
(getDeadlineForNextIdleCallback):
(getRICRetryCount):
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/requestidlecallback/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/266678@main
